src/main/scala/io/iohk/ethereum/blockchain/sync/RegularSyncController.scala

```diff
@@ -0,0 +1,181 @@
+package io.iohk.ethereum.blockchain.sync
+
+import akka.actor.{Actor, ActorContext, ActorRef, Scheduler}
+import akka.event.LoggingAdapter
+import akka.util.ByteString
+import io.iohk.ethereum.blockchain.sync.FastSyncController.StartRegularSync
+import io.iohk.ethereum.db.storage.AppStateStorage
+import io.iohk.ethereum.domain.{Block, BlockHeader, Blockchain}
+import io.iohk.ethereum.network.PeerActor.{BroadcastBlocks, MessageReceived, SendMessage, Subscribe}
+import io.iohk.ethereum.network.p2p.messages.CommonMessages.{NewBlock, Status}
+import io.iohk.ethereum.network.p2p.messages.PV62._
+import io.iohk.ethereum.network.p2p.validators.BlockValidator.BlockError
+import io.iohk.ethereum.utils.Config
+
+import scala.concurrent.ExecutionContext.Implicits.global
+import scala.concurrent.duration._
+import scala.util.Try
+
+trait RegularSyncController {
+
+  val appStateStorage: AppStateStorage
+  val blockchain: Blockchain
+  val context: ActorContext
+  val blockValidator: (BlockHeader, BlockBody) => Either[BlockError, Block]
+
+  def blacklist(peer: ActorRef, duration: FiniteDuration)
+
+  def peersToDownloadFrom: Map[ActorRef, Status]
+
+  def scheduler: Scheduler
+
+  private var headers: Seq[BlockHeader] = Seq.empty
+//  private var totalDifficulty: Option[BigInt] = None
+  private var resolvingBranch = false
+  private var broadcasting = false
+
+  import Config.FastSync._
+
+  private implicit val sender: ActorRef = context.self
+
+  def regularSync(log: LoggingAdapter): Actor.Receive = {
```

Comment:
I add self, it turn out that this was the fix for tests
https://github.com/input-output-hk/etc-client/pull/77/commits/a22aaaddfa0cf617694033ad2ed7117381712434